ext/npm-extension.js

```diff
@@ -258,6 +258,11 @@ exports.addExtension = function(System){
 	System.locate = function(load){
 		var parsedModuleName = utils.moduleName.parse(load.name),
 			loader = this;
+
+		if (load.metadata) {
+			load.metadata.parsedModuleName = parsedModuleName;
```

Comment:
The test below fails without the conditional, `metadata` is not set for some reason. Want me to figure out why? Should `metadata` always be available at that point?

https://github.com/stealjs/steal/blob/915c636bef2018e7d3a30e2aaf8855b4c554c3aa/test/npm/test.js#L165-L185